test/parallel/test-module-loading-error.js

```diff
@@ -84,10 +84,15 @@ assert.throws(
     message: 'The argument \'id\' must be a non-empty string. Received \'\''
   });
 
-assert.throws(
-  () => { require('../fixtures/packages/is-dir'); },
-  {
-    code: 'MODULE_NOT_FOUND',
-    message: /Cannot find module '\.\.\/fixtures\/packages\/is-dir'/
-  }
-);
+// Folder read operation succeeds in AIX.
+// For libuv change, see https://github.com/libuv/libuv/pull/2025.
+// https://github.com/nodejs/node/pull/48477#issuecomment-1604586650
+if (process.platform !== 'aix') {
```

Comment:
Let's validate the behavior on AIX so we can spot potential regressions.
```suggestion
if (process.platform === 'aix') {
  require('../fixtures/packages/is-dir');
} else {
```

Reply:
If we are testing, we should test for the error message, since it throws `Error parsing` error here. Since we don't test error messages mostly, should we still test it?

```
15:23:18 not ok 2008 parallel/test-module-loading-error
15:23:18   ---
15:23:18   duration_ms: 230.33100
15:23:18   severity: fail
15:23:18   exitcode: 1
15:23:18   stack: |-
15:23:18     node:assert:635
15:23:18           throw err;
15:23:18           ^
15:23:18     
15:23:18     AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
15:23:18     + actual - expected
15:23:18     
15:23:18       Comparison {
15:23:18     +   message: `Error parsing /home/iojs/build/workspace/node-test-commit-aix/nodes/aix72-ppc64/test/fixtures/packages/is-dir/package.json: Unexpected token 'U', "Uï¿½.\x00\x00\x00\x00\x00\x00\x00"... is not valid JSON`
15:23:18     -   code: 'MODULE_NOT_FOUND',
15:23:18     -   message: /Cannot find module '\.\.\/fixtures\/packages\/is-dir'/
15:23:18       }
15:23:18         at Object.<anonymous> (/home/iojs/build/workspace/node-test-commit-aix/nodes/aix72-ppc64/test/parallel/test-module-loading-error.js:87:8)
15:23:18         at Module._compile (node:internal/modules/cjs/loader:1233:14)
15:23:18         at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
15:23:18         at Module.load (node:internal/modules/cjs/loader:1091:32)
15:23:18         at Module._load (node:internal/modules/cjs/loader:938:12)
15:23:18         at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
15:23:18         at node:internal/main/run_main_module:23:47 {
15:23:18       generatedMessage: true,
15:23:18       code: 'ERR_ASSERTION',
15:23:18       actual: SyntaxError: Error parsing /home/iojs/build/workspace/node-test-commit-aix/nodes/aix72-ppc64/test/fixtures/packages/is-dir/package.json: Unexpected token 'U', "Uï¿½."... is not valid JSON
15:23:18           at parse (<anonymous>)
15:23:18           at Object.read (node:internal/modules/package_json_reader:63:16)
15:23:18           at readPackage (node:internal/modules/cjs/loader:362:28)
15:23:18           at tryPackage (node:internal/modules/cjs/loader:401:15)
15:23:18           at Module._findPath (node:internal/modules/cjs/loader:665:18)
15:23:18           at Module._resolveFilename (node:internal/modules/cjs/loader:1034:27)
15:23:18           at Module._load (node:internal/modules/cjs/loader:901:27)
15:23:18           at Module.require (node:internal/modules/cjs/loader:1115:19)
15:23:18           at require (node:internal/modules/helpers:121:18)
15:23:18           at assert.throws.code (/home/iojs/build/workspace/node-test-commit-aix/nodes/aix72-ppc64/test/parallel/test-module-loading-error.js:88:11) {
15:23:18         path: '/home/iojs/build/workspace/node-test-commit-aix/nodes/aix72-ppc64/test/fixtures/packages/is-dir/package.json'
15:23:18       },
15:23:18       expected: {
15:23:18         code: 'MODULE_NOT_FOUND',
15:23:18         message: /Cannot find module '\.\.\/fixtures\/packages\/is-dir'/
15:23:18       },
15:23:18       operator: 'throws'
15:23:18     }
15:23:18     
15:23:18     Node.js v21.0.0-pre
15:23:18   ...
```

Reply:
Do you know what this error message means? Like is there a virtual file or something like that that Node.js tries to parse or is that just random data, or what? I think that would quite confusing for users on AIX, I wonder if we could improve that somehow 🤔

Reply:
@richardlau commented on the previous pr: https://github.com/nodejs/node/pull/48477#issuecomment-1604586650

Instead of throwing EISDIR error, it just succeeds, and because it succeeds, JS land `JSONParse()` throws error, which throws the `Error parsing...` error message.

Reply:
But in any case, yes we should have _some test_, ideally with an error code – which is another breaking change, but this PR is semver-major anyway.

Reply:
We have another TODO in `package-json-reader.js` to remove the error differences in different loaders. I can leave this PR as an error message check, and in another PR (which is going to be semver-major as well), we can get rid of the `Error parsing...` error, and use `ERR_INVALID_PACKAGE_CONFIG` error code.

Reply:
I think it would make more sense to land the other PR first. I’ll try to open that tomorrow, or you can beat me to it if you prefer

Reply:
I've updated the PR